crates/host/src/wasmbus/mod.rs

```diff
@@ -2000,13 +1993,84 @@ impl Host {
+                                            trace!(provider_id=health_provider_id, "provider health check succeeded");
+                                            previous_healthy = true;
+                                            if let Err(e) = event::publish(
+                                                &event_builder,
+                                                &ctl_nats,
+                                                &health_lattice_prefix,
+                                                "health_check_passed",
+                                                event::provider_health_check(
+                                                    &health_provider_id,
+                                                    &health_link_name,
+                                                    &health_contract_id,
+                                                )
+                                            ).await {
+                                                warn!(?e, "failed to publish provider health check succeeded event");
+                                            }
+                                        },
+                                        (Ok(HealthCheckResponse { healthy: false, ..}), true) => {
+                                            trace!(provider_id=health_provider_id, "provider health check failed");
+                                            previous_healthy = false;
+                                            if let Err(e) = event::publish(
+                                                &event_builder,
+                                                &ctl_nats,
+                                                &health_lattice_prefix,
+                                                "health_check_failed",
+                                                event::provider_health_check(
+                                                    &health_provider_id,
+                                                    &health_link_name,
+                                                    &health_contract_id,
+                                                )
+                                            ).await {
+                                                warn!(?e, "failed to publish provider health check failed event");
+                                            }
+                                        }
+                                        // If the provider health status didn't change, we don't need to publish an event
+                                        (Ok(_), _) => (),
+                                        _ => warn!("failed to deserialize provider health check response"),
+                                    }
+                                }
+                                else {
+                                    warn!("failed to request provider health, retrying in 30 seconds");
+                                }
+                        }
```

Comment:
There's way too much nesting here. I'm fine with this if we add a `TODO` to refactor, otherwise, could we adapt an approach similar to https://github.com/wasmCloud/wasmCloud/blob/b77767e6d3c32ceba0b4e5b421b532ac0788dc15/tests/wasmbus.rs#L96-L115 perhaps?

Reply:
I'll add a TODO here so we can tackle it, along with the idea of cloning Arc<Self> to clean up the code